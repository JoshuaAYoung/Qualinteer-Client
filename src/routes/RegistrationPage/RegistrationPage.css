
.registerButton {
  box-sizing: border-box;
  display: inline-block;
  font-size: 1rem;
  padding: 0.35em 1.2em;
  border: 0.1em solid black;
  border-radius: 0.2em;
  margin: 10px 8px 30px 0;
  transition: all 0.6s;
  background-color: transparent;
}

.registerButton:hover {
  color: #000000;
  background-color: lightgray;
}

.registerButton:disabled, .registerButton[disabled] {
  color: gray;
  background-color: red;
  border: 0.1em solid gray;
}

.registerLabel {
  display: inline-block;
  margin-bottom: 10px;
  font-weight: 500;
  font-size: 1.2em;
}

.registerInput {
  font-size: 1.2em;
  width: 420px;
}

@media screen and (max-width: 480px) {
  .registerInput {
    width: 100%;
  }
  
}