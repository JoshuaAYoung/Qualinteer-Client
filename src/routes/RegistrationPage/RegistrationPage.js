import React, { Component } from 'react';
import OrganizationRegistration from '../../components/RegistrationForms/OrganizationRegistration'
import VolunteerRegistration from '../../components/RegistrationForms/VolunteerRegistration'
import './RegistrationPage.css'
import AuthApiService from '../../services/auth-api-service'
import ErrorToast from '../../components/ErrorToast/ErrorToast'
import ScrollToTopOnMount from '../../utilities/ScrollToTopOnMount'

export default class RegistrationPage extends Component {
  state = {
    error: null,
    header: '',
    emailTempValue: { value: '', touched: false },
    passwordTempValue: { value: '', touched: false },
    full_nameTempValue: '',
    user_typeTempValue: '',
    nameTempValue: '',
    addressTempValue: '',
    cityTempValue: '',
    stateTempValue: '',
    zipcodeTempValue: { value: '', touched: false },
    phoneTempValue: '',
    websiteTempValue: '',
  }

  addTempValue = (field, input) => {
    this.setState({
      [field + 'TempValue']: input
    })
  }

  addTempValidation = (field, input) => {
    this.setState({
      [field + 'TempValue']: {
        value: input,
        touched: true
      }
    })
  }

  handleSubmit = ev => {
    ev.preventDefault()
    let allFields
    const userType = this.props.match.params.user
    const userFields = {
      email: this.state.emailTempValue.value,
      password: this.state.passwordTempValue.value,
      full_name: this.state.full_nameTempValue,
      user_type: userType
    }
    if (userType === 'organization') {
      const orgFields = {
        name: this.state.nameTempValue,
        address: this.state.addressTempValue,
        city: this.state.cityTempValue,
        state: this.state.stateTempValue,
        zipcode: this.state.zipcodeTempValue.value,
        phone: this.state.phoneTempValue,
        website: this.state.websiteTempValue
      }
      allFields = { ...userFields, ...orgFields }
    }
    else {
      allFields = { ...userFields }
    }

    this.setState({ error: null })
    AuthApiService.postUser(allFields)
      .then(user => {
        this.props.history.push('/success/login')
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  generateError = () => {
    const { emailTempValue, passwordTempValue, zipcodeTempValue } = this.state;
    //eslint-disable-next-line
    const passRegEx = /^(?:(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).*)$/;
    //eslint-disable-next-line
    const emailRegEx = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
    const zipcodeRegEx = /\b\d{5}\b/;
    const errorProps = {};
    if (this.state.emailTempValue.touched === true && !emailRegEx.test(emailTempValue.value)) {
      errorProps.emailError = 'Please enter a valid email address.';
    }
    if (this.state.passwordTempValue.touched === true && !passRegEx.test(passwordTempValue.value)) {
      errorProps.passwordError1 = 'Password must contain an upper case letter, a lower case letter and a number.';
    }
    if (this.state.passwordTempValue.touched === true && passwordTempValue.value.length < 8) {
      errorProps.passwordError2 = 'Password must be equal to or longer than 8 characters.';
    }
    if (this.state.passwordTempValue.touched === true && passwordTempValue.value.length > 71) {
      errorProps.passwordError3 = 'Password must be less than 72 characters.';
    }
    if (this.state.passwordTempValue.touched === true && (passwordTempValue.value.startsWith(' ') || passwordTempValue.value.endsWith(' '))) {
      errorProps.passwordError4 = 'Password must not start or end with empty spaces.';
    }
    if (this.state.zipcodeTempValue.touched === true && (zipcodeTempValue.value.length !== 5 || !zipcodeRegEx.test(zipcodeTempValue.value))) {
      errorProps.zipcodeError = 'Zipcode must be a 5 digit number.';
    }
    return errorProps;
  }

  componentDidMount() {
    if (this.props.match.params.user === 'organization') {
      this.setState({ header: 'Find Help.' })
    }
    else if (this.props.match.params.user === 'volunteer') {
      this.setState({ header: 'Volunteer.' })
    }
  }

  formRender = () => {
    if (this.props.match.params.user === 'organization') {
      return (
        <OrganizationRegistration
          addTempValidation={this.addTempValidation}
          addTempValue={this.addTempValue}
          errorProps={this.generateError()}
        />
      )
    }
    else if (this.props.match.params.user === 'volunteer') {
      return (
        <VolunteerRegistration
          addTempValidation={this.addTempValidation}
          addTempValue={this.addTempValue}
          errorProps={this.generateError()}
        />
      )
    }
  }

  render() {
    const { error } = this.state
    return (
      <div>
        <ScrollToTopOnMount />
        <section className='registrationContainer'>
          <div className="registerBanner">
            <h1 className='pageHeader'>{this.state.header}</h1>
            <p className='pageInstructions'>All fields are required unless noted otherwise.</p>
          </div>
          {error && <ErrorToast errorMessage={error} />}
          <form className='signupForm' onSubmit={ev => this.handleSubmit(ev)}>
            {this.formRender()}
            <button
              type='submit'
              className='registerButton'
              disabled={Object.keys(this.generateError()).length}
            >
              Sign Up
              </button>
          </form>
        </section >
      </div >
    );
  }
}
