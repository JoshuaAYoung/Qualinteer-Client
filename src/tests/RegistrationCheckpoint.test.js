import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationPage from '../routes/RegistrationPage/RegistrationPage';


describe('RegistrationPage Component', () => {
  const props = {
    match: {
      params: {
        user: 'volunteer'
      }
    }
  }

  it('RegistrationPage Smoke Test', () => {
    const div = document.createElement('div')

    ReactDOM.render(<RegistrationPage {...props} />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  it('RegistrationPage Snapshot Test', () => {
    const wrapper = shallow(<RegistrationPage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})