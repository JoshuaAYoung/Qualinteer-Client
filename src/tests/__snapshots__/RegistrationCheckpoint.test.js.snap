// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationPage Component RegistrationPage Snapshot Test 1`] = `
<div>
  <ScrollToTopOnMount />
  <section
    className="registrationContainer"
  >
    <div
      className="registerBanner"
    >
      <h1
        className="pageHeader"
      >
        Volunteer.
      </h1>
      <p
        className="pageInstructions"
      >
        All fields are required unless noted otherwise.
      </p>
    </div>
    <form
      className="signupForm"
      onSubmit={[Function]}
    >
      <VolunteerRegistration
        addTempValidation={[Function]}
        addTempValue={[Function]}
        errorProps={Object {}}
      />
      <button
        className="registerButton"
        disabled={0}
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </section>
</div>
`;
